src/cs3iface.py

```diff
@@ -9,137 +9,215 @@
 '''
 
 import time
-import os
+import http
+import requests
 import grpc
 
-import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
+#from tusclient import client as tusclient
+
+#from google.auth.transport import grpc as google_auth_transport_grpc
+#from google.auth import jwt as google_auth_jwt
+#from google import auth as google_auth
+
+import cs3.storage.provider.v1beta1.resources_pb2 as spr
+import cs3.storage.provider.v1beta1.provider_api_pb2 as sp
 import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
+import cs3.rpc.code_pb2 as cs3code
+
 
 # module-wide state
-config = None
-log = None
-tok = None
+ctx = {}        # "map" to store some module context: cf. init()
+tokens = {}     # map userid [string] to authentication token
 
-def _getfilename(filename):
-  '''map the given filename into the target namespace by prepending the homepath (see storagehomepath in wopiserver.conf)'''
-  return filename   # TODO do we need that??
 
-def init(inconfig, inlog):
+def init(config, log):
   '''Init module-level variables'''
-  global config         # pylint: disable=global-statement
-  global log            # pylint: disable=global-statement
-  global tok            # pylint: disable=global-statement
-  config = inconfig
-  log = inlog
-  #revaurl = config.get('cs3', 'revaurl')
-  revaurl = 'iota:19000' # XXX for now
-
+  ctx['log'] = log
+  ctx['chunksize'] = config.getint('io', 'chunksize')
+  revahost = config.get('cs3', 'revahost')
   # prepare the gRPC connection
-  ch = grpc.insecure_channel(revaurl)
-  cs3stub = cs3gw_grpc.GatewayAPIStub(ch)
-  authReq = cs3gw.AuthenticateRequest(type='basic', client_id='gonzalhu', client_secret='test')
-  tok = cs3stub.Authenticate(authReq)
-
-
-def stat(_endpoint, filename, _ruid, _rgid):
-  '''Stat a file and returns (size, mtime) as well as other extended info. Assume the given uid, gid has access.'''
-  filename = _getfilename(filename)
-  try:
-    tstart = time.clock()
-    statInfo = os.stat(filename)
-    tend = time.clock()
-    log.info('msg="Invoked stat" filename="%s" elapsedTimems="%.1f"' % (filename, (tend-tstart)*1000))
+  ch = grpc.insecure_channel(revahost)
+  ctx['cs3stub'] = cs3gw_grpc.GatewayAPIStub(ch)
+
+
+def _authenticate(userid):
+  '''Obtain a token from Reva for the given userid'''
+  # TODO this will become
+  #authReq = cs3gw.AuthenticateRequest(type='bearer', client_secret=userid)
```

Comment:
I created a ticket for this TODO since I guess it's rather important. https://github.com/sciencemesh/sciencemesh/issues/118

Reply:
Indeed, as also discussed with @ishank011. We need to start using proper authentication, and adapt the test suite accordingly.

Reply:
But let's do that in anothet PR